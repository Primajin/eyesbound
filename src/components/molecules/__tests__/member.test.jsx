import React from 'react';
import {act, create} from 'react-test-renderer';

import Member from '../member.jsx';

describe('Member', () => {
	it('renders correctly with a list of zero', () => {
		const name = 'name';
		const title = 'title';
		const component = create(<Member edges={[]} title={title} name={name}/>);
		act(() => {
			component.root.findByType('button').props.onClick();
		});
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with a list of one', () => {
		const picture = {coordinates: {latitude: 52.123, longitude: 13.123}, image: {}, title: 'title1'};
		const thumbnailData = {data: picture, id: 'id-foo', uid: 'uid-bar'};
		const thumbnailDataNode = {node: thumbnailData};
		const edges = [thumbnailDataNode];
		const name = 'name';
		const title = 'title';
		const component = create(<Member edges={edges} title={title} name={name}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with a list of more', () => {
		const picture1 = {coordinates: {latitude: 52.456, longitude: 13.456}, image: {}, title: 'title1'};
		const picture2 = {coordinates: {latitude: 52.789, longitude: 13.798}, image: {}, title: 'title2'};
		const thumbnailData1 = {data: picture1, id: 'id-1', uid: 'uid-1'};
		const thumbnailData2 = {data: picture2, id: 'id-2', uid: 'uid-2'};
		const thumbnailDataNode1 = {node: thumbnailData1};
		const thumbnailDataNode2 = {node: thumbnailData2};
		const edges = [thumbnailDataNode1, thumbnailDataNode2];
		const name = 'name';
		const title = 'title';
		const component = create(<Member edges={edges} title={title} name={name}/>);
		expect(component).toMatchSnapshot();
	});
});
