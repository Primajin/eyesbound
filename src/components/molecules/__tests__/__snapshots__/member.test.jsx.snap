// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member renders correctly with a list of more 1`] = `
[
  <div
    className="mock-helmet"
  >
    <title>
      title | EYESBOUND
    </title>
    <meta
      content="title | EYESBOUND"
      name="title"
    />
    <meta
      content="title | EYESBOUND"
      property="og:title"
    />
    <meta
      content="title | EYESBOUND"
      name="twitter:title"
    />
  </div>,
  <header>
    <h1
      className=""
      css={
        {
          "name": "1vze745",
          "next": undefined,
          "styles": "
	background-color: var(--foreground);
	color: var(--background);
	font-size: 27px;
	font-weight: 700;
	height: 100px;
	left: 0;
	line-height: 100px;
	opacity: 1;
	position: fixed;
	text-align: center;
	transition: opacity .333s;
	top: 0;
	width: 250px;
	z-index: 10;

	&.isFullscreen {
		opacity: 0.65;
	}

	@media (min-width: 576px) {
		font-size: 37px;
		width: 400px;
	};
",
        }
      }
    >
      <a
        href="/"
        tabIndex="1"
      >
        EYESBOUND
      </a>
    </h1>
    <nav
      className=""
      css={
        {
          "name": "1ghowhq",
          "next": undefined,
          "styles": "
	line-height: 40px;
	opacity: 1;
	outline: 0;
	position: fixed;
	text-align: center;
	top: 100px;
	transform: translateY(calc(-100% + 40px));
	transition: transform .333s, opacity .333s;
	user-select: none;
	width: 180px;
	z-index: 9;

	@media (min-width: 768px) {
		transform: translateY(calc(-100% + 20px));
	};

	&.isFullscreen {
		opacity: 0;
	}

	div {
		transition: background-color .333s, color .333s;
		background-color: var(--background);
		color: var(--foreground);

		@media (min-width: 768px) {
			line-height: 20px;
		};

		&:hover {
			background: var(--foreground);
			color: var(--background);
		}
	}

	&.open {
		transform: translateY(0);

		div {
			cursor: default;

			&:hover {
				background-color: var(--background);
				color: var(--foreground);
			}
		}
	}

	ul {
		background-color: var(--foreground);
		color: var(--background);
	}

	li {
		transition: opacity .333s;

		&:hover {
			opacity: 0.666;
		}
	}

	div, li {
		cursor: pointer;
		font-weight: 700;
		text-transform: uppercase;
	}
",
        }
      }
    >
      <ul>
        <li>
          <a
            href="/picture"
            tabIndex="2"
          >
            Overview
          </a>
        </li>
        <li>
          <a
            href="/series/shuttered"
            tabIndex="3"
          >
            »Shuttered«
          </a>
        </li>
        <li>
          <a
            href="/category/architecture"
            tabIndex="4"
          >
            Architecture
          </a>
        </li>
        <li>
          <a
            href="/category/environment"
            tabIndex="5"
          >
            Environment
          </a>
        </li>
        <li>
          <a
            href="/category/floral"
            tabIndex="6"
          >
            Floral
          </a>
        </li>
        <li>
          <a
            href="/category/light"
            tabIndex="7"
          >
            Light
          </a>
        </li>
        <li>
          <a
            href="/worldmap"
            tabIndex="8"
          >
            Worldmap
          </a>
        </li>
        <li>
          <a
            href="/imprint"
            tabIndex="9"
          >
            Contact
          </a>
        </li>
      </ul>
      <div
        onClick={[Function]}
      >
        Menu
      </div>
    </nav>
    <button
      aria-label="Switch theme"
      className=""
      css={
        {
          "name": "jwavk5",
          "next": undefined,
          "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
        }
      }
      onClick={[Function]}
      type="button"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 21c0 .6.5 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .6.5 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2zm2.9 11.1l-.9.6V16h-4v-2.3l-.8-.6a5 5 0 115.7 0z"
        />
      </svg>
      <div
        className="mock-helmet"
      >
        <meta
          content="#000000"
          name="theme-color"
        />
      </div>
    </button>
  </header>,
  <main
    css={
      {
        "name": "14akp75",
        "next": undefined,
        "styles": "
	width: clamp(16rem, 90vw, 70rem);
	margin: 170px auto 100px;
	padding: 0 1.5rem;

	h1, h3 {
		font-weight: 400;
		padding-bottom: 1em;
	}
",
      }
    }
  >
    <h1>
      name
      : 
      title
       (
      2
      )
    </h1>
    <ul
      css={
        {
          "name": "1ktsphj",
          "next": undefined,
          "styles": "
	display: grid;
	grid-template-columns: 1fr;
	gap: 5px;

	@media (min-width: 420px) {
		grid-template-columns: 1fr 1fr;
	};

	@media (min-width: 576px) {
		grid-template-columns: 1fr 1fr 1fr;
    gap: 10px;
	};

	@media (min-width: 992px) {
		grid-template-columns: 1fr 1fr 1fr 1fr;
		gap: 15px;
	};

	li {
		content-visibility: auto;
	}
",
        }
      }
    >
      <li>
        <a
          aria-label="link-to-picture"
          href="/picture/uid-1"
        >
          <picture>
            <img
              alt="title1"
              css={
                {
                  "name": "m2yydh",
                  "next": undefined,
                  "styles": "
	padding-top: 66.73913043478261%;
",
                }
              }
              height={174}
              image={
                {
                  "layout": "CONSTRAINED",
                }
              }
              width={261}
            />
          </picture>
        </a>
      </li>
      <li>
        <a
          aria-label="link-to-picture"
          href="/picture/uid-2"
        >
          <picture>
            <img
              alt="title2"
              css={
                {
                  "name": "m2yydh",
                  "next": undefined,
                  "styles": "
	padding-top: 66.73913043478261%;
",
                }
              }
              height={174}
              image={
                {
                  "layout": "CONSTRAINED",
                }
              }
              width={261}
            />
          </picture>
        </a>
      </li>
    </ul>
  </main>,
]
`;

exports[`Member renders correctly with a list of one 1`] = `
[
  <div
    className="mock-helmet"
  >
    <title>
      title | EYESBOUND
    </title>
    <meta
      content="title | EYESBOUND"
      name="title"
    />
    <meta
      content="title | EYESBOUND"
      property="og:title"
    />
    <meta
      content="title | EYESBOUND"
      name="twitter:title"
    />
  </div>,
  <header>
    <h1
      className=""
      css={
        {
          "name": "1vze745",
          "next": undefined,
          "styles": "
	background-color: var(--foreground);
	color: var(--background);
	font-size: 27px;
	font-weight: 700;
	height: 100px;
	left: 0;
	line-height: 100px;
	opacity: 1;
	position: fixed;
	text-align: center;
	transition: opacity .333s;
	top: 0;
	width: 250px;
	z-index: 10;

	&.isFullscreen {
		opacity: 0.65;
	}

	@media (min-width: 576px) {
		font-size: 37px;
		width: 400px;
	};
",
        }
      }
    >
      <a
        href="/"
        tabIndex="1"
      >
        EYESBOUND
      </a>
    </h1>
    <nav
      className=""
      css={
        {
          "name": "1ghowhq",
          "next": undefined,
          "styles": "
	line-height: 40px;
	opacity: 1;
	outline: 0;
	position: fixed;
	text-align: center;
	top: 100px;
	transform: translateY(calc(-100% + 40px));
	transition: transform .333s, opacity .333s;
	user-select: none;
	width: 180px;
	z-index: 9;

	@media (min-width: 768px) {
		transform: translateY(calc(-100% + 20px));
	};

	&.isFullscreen {
		opacity: 0;
	}

	div {
		transition: background-color .333s, color .333s;
		background-color: var(--background);
		color: var(--foreground);

		@media (min-width: 768px) {
			line-height: 20px;
		};

		&:hover {
			background: var(--foreground);
			color: var(--background);
		}
	}

	&.open {
		transform: translateY(0);

		div {
			cursor: default;

			&:hover {
				background-color: var(--background);
				color: var(--foreground);
			}
		}
	}

	ul {
		background-color: var(--foreground);
		color: var(--background);
	}

	li {
		transition: opacity .333s;

		&:hover {
			opacity: 0.666;
		}
	}

	div, li {
		cursor: pointer;
		font-weight: 700;
		text-transform: uppercase;
	}
",
        }
      }
    >
      <ul>
        <li>
          <a
            href="/picture"
            tabIndex="2"
          >
            Overview
          </a>
        </li>
        <li>
          <a
            href="/series/shuttered"
            tabIndex="3"
          >
            »Shuttered«
          </a>
        </li>
        <li>
          <a
            href="/category/architecture"
            tabIndex="4"
          >
            Architecture
          </a>
        </li>
        <li>
          <a
            href="/category/environment"
            tabIndex="5"
          >
            Environment
          </a>
        </li>
        <li>
          <a
            href="/category/floral"
            tabIndex="6"
          >
            Floral
          </a>
        </li>
        <li>
          <a
            href="/category/light"
            tabIndex="7"
          >
            Light
          </a>
        </li>
        <li>
          <a
            href="/worldmap"
            tabIndex="8"
          >
            Worldmap
          </a>
        </li>
        <li>
          <a
            href="/imprint"
            tabIndex="9"
          >
            Contact
          </a>
        </li>
      </ul>
      <div
        onClick={[Function]}
      >
        Menu
      </div>
    </nav>
    <button
      aria-label="Switch theme"
      className=""
      css={
        {
          "name": "jwavk5",
          "next": undefined,
          "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
        }
      }
      onClick={[Function]}
      type="button"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2z"
        />
      </svg>
      <div
        className="mock-helmet"
      >
        <meta
          content="#ffffff"
          name="theme-color"
        />
      </div>
    </button>
  </header>,
  <main
    css={
      {
        "name": "14akp75",
        "next": undefined,
        "styles": "
	width: clamp(16rem, 90vw, 70rem);
	margin: 170px auto 100px;
	padding: 0 1.5rem;

	h1, h3 {
		font-weight: 400;
		padding-bottom: 1em;
	}
",
      }
    }
  >
    <h1>
      name
      : 
      title
       (
      1
      )
    </h1>
    <ul
      css={
        {
          "name": "1ktsphj",
          "next": undefined,
          "styles": "
	display: grid;
	grid-template-columns: 1fr;
	gap: 5px;

	@media (min-width: 420px) {
		grid-template-columns: 1fr 1fr;
	};

	@media (min-width: 576px) {
		grid-template-columns: 1fr 1fr 1fr;
    gap: 10px;
	};

	@media (min-width: 992px) {
		grid-template-columns: 1fr 1fr 1fr 1fr;
		gap: 15px;
	};

	li {
		content-visibility: auto;
	}
",
        }
      }
    >
      <li>
        <a
          aria-label="link-to-picture"
          href="/picture/uid-bar"
        >
          <picture>
            <img
              alt="title1"
              css={
                {
                  "name": "m2yydh",
                  "next": undefined,
                  "styles": "
	padding-top: 66.73913043478261%;
",
                }
              }
              height={174}
              image={
                {
                  "layout": "CONSTRAINED",
                }
              }
              width={261}
            />
          </picture>
        </a>
      </li>
    </ul>
  </main>,
]
`;

exports[`Member renders correctly with a list of zero 1`] = `
[
  <div
    className="mock-helmet"
  >
    <title>
      title | EYESBOUND
    </title>
    <meta
      content="title | EYESBOUND"
      name="title"
    />
    <meta
      content="title | EYESBOUND"
      property="og:title"
    />
    <meta
      content="title | EYESBOUND"
      name="twitter:title"
    />
  </div>,
  <header>
    <h1
      className=""
      css={
        {
          "name": "1vze745",
          "next": undefined,
          "styles": "
	background-color: var(--foreground);
	color: var(--background);
	font-size: 27px;
	font-weight: 700;
	height: 100px;
	left: 0;
	line-height: 100px;
	opacity: 1;
	position: fixed;
	text-align: center;
	transition: opacity .333s;
	top: 0;
	width: 250px;
	z-index: 10;

	&.isFullscreen {
		opacity: 0.65;
	}

	@media (min-width: 576px) {
		font-size: 37px;
		width: 400px;
	};
",
        }
      }
    >
      <a
        href="/"
        tabIndex="1"
      >
        EYESBOUND
      </a>
    </h1>
    <nav
      className=""
      css={
        {
          "name": "1ghowhq",
          "next": undefined,
          "styles": "
	line-height: 40px;
	opacity: 1;
	outline: 0;
	position: fixed;
	text-align: center;
	top: 100px;
	transform: translateY(calc(-100% + 40px));
	transition: transform .333s, opacity .333s;
	user-select: none;
	width: 180px;
	z-index: 9;

	@media (min-width: 768px) {
		transform: translateY(calc(-100% + 20px));
	};

	&.isFullscreen {
		opacity: 0;
	}

	div {
		transition: background-color .333s, color .333s;
		background-color: var(--background);
		color: var(--foreground);

		@media (min-width: 768px) {
			line-height: 20px;
		};

		&:hover {
			background: var(--foreground);
			color: var(--background);
		}
	}

	&.open {
		transform: translateY(0);

		div {
			cursor: default;

			&:hover {
				background-color: var(--background);
				color: var(--foreground);
			}
		}
	}

	ul {
		background-color: var(--foreground);
		color: var(--background);
	}

	li {
		transition: opacity .333s;

		&:hover {
			opacity: 0.666;
		}
	}

	div, li {
		cursor: pointer;
		font-weight: 700;
		text-transform: uppercase;
	}
",
        }
      }
    >
      <ul>
        <li>
          <a
            href="/picture"
            tabIndex="2"
          >
            Overview
          </a>
        </li>
        <li>
          <a
            href="/series/shuttered"
            tabIndex="3"
          >
            »Shuttered«
          </a>
        </li>
        <li>
          <a
            href="/category/architecture"
            tabIndex="4"
          >
            Architecture
          </a>
        </li>
        <li>
          <a
            href="/category/environment"
            tabIndex="5"
          >
            Environment
          </a>
        </li>
        <li>
          <a
            href="/category/floral"
            tabIndex="6"
          >
            Floral
          </a>
        </li>
        <li>
          <a
            href="/category/light"
            tabIndex="7"
          >
            Light
          </a>
        </li>
        <li>
          <a
            href="/worldmap"
            tabIndex="8"
          >
            Worldmap
          </a>
        </li>
        <li>
          <a
            href="/imprint"
            tabIndex="9"
          >
            Contact
          </a>
        </li>
      </ul>
      <div
        onClick={[Function]}
      >
        Menu
      </div>
    </nav>
    <button
      aria-label="Switch theme"
      className=""
      css={
        {
          "name": "jwavk5",
          "next": undefined,
          "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
        }
      }
      onClick={[Function]}
      type="button"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 21c0 .6.5 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .6.5 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2zm2.9 11.1l-.9.6V16h-4v-2.3l-.8-.6a5 5 0 115.7 0z"
        />
      </svg>
      <div
        className="mock-helmet"
      >
        <meta
          content="#000000"
          name="theme-color"
        />
      </div>
    </button>
  </header>,
  <main
    css={
      {
        "name": "14akp75",
        "next": undefined,
        "styles": "
	width: clamp(16rem, 90vw, 70rem);
	margin: 170px auto 100px;
	padding: 0 1.5rem;

	h1, h3 {
		font-weight: 400;
		padding-bottom: 1em;
	}
",
      }
    }
  >
    <h1>
      name
      : 
      title
       (
      0
      )
    </h1>
    <ul
      css={
        {
          "name": "1ktsphj",
          "next": undefined,
          "styles": "
	display: grid;
	grid-template-columns: 1fr;
	gap: 5px;

	@media (min-width: 420px) {
		grid-template-columns: 1fr 1fr;
	};

	@media (min-width: 576px) {
		grid-template-columns: 1fr 1fr 1fr;
    gap: 10px;
	};

	@media (min-width: 992px) {
		grid-template-columns: 1fr 1fr 1fr 1fr;
		gap: 15px;
	};

	li {
		content-visibility: auto;
	}
",
        }
      }
    />
  </main>,
]
`;
